client_http.hpp

```diff
@@ -12,15 +11,40 @@
 #include <mutex>
 #include <type_traits>
 
+#ifdef USE_BOOST_ASIO
+#include <boost/asio.hpp>
+#define ASIO_NS boost::asio
```

Comment:
Pretty much to be style consistant with [REGEX_NS used on the server side](https://github.com/eidheim/Simple-Web-Server/blob/master/server_http.hpp#L38).